.github/workflows/tech-debt-burndown.yml

```
@@ -0,0 +1,139 @@
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        with:
+          ref: main
+        if: ${{ matrix.branch == 'main' }}
+
+      - name: Checkout PR branch
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        if: ${{ matrix.branch == 'pr' }}
+
+      - name: Get Core Dependencies
+        uses: ./.github/workflows/actions/get-core-dependencies
+
+      - name: Install tsc-output-parser
+        run: npm install @aivenio/tsc-output-parser@2.1.1
+
+      - name: Run Typescript compiler and generate JSON-formatted error file
+        run: npx tsc --strictNullChecks --noEmit --pretty false | npx tsc-output-parser > null_errors_${{ matrix.branch }}.json
+
+      - name: Upload null_errors_${{ matrix.branch }}.json
+        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535 # v3.0.0
+        with:
+          name: null_errors_${{ matrix.branch }}
+          path: 'null_errors_${{ matrix.branch }}.json'
+
+  # TODO(STENCIL-454): Remove or change this up once we've eliminated unused exports
+  unused_exports_check:
+    strategy:
+      matrix:
+        branch: [ 'main', 'pr' ]
+    name: Find unused variables on ${{ matrix.branch }}
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        with:
+          ref: main
+        if: ${{ matrix.branch == 'main' }}
+
+      - name: Checkout PR branch
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        if: ${{ matrix.branch == 'pr' }}
+
+      - name: Install ts-prune
+        run: npm install ts-prune@0.10.3
+
+      - name: Run ESLint and write output to disk
+        run: npx ts-prune > unused-exports-${{ matrix.branch }}.txt
+
+      - name: Upload unused exports
+        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535 # v3.0.0
+        with:
+          name: unused-exports-${{ matrix.branch }}
+          path: 'unused-exports-${{ matrix.branch }}.txt'
+
+  format_report:
+    needs: [ "strict_null_check", "unused_exports_check" ]
+    name: Download error files and report
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Get Core Dependencies
+        uses: ./.github/workflows/actions/get-core-dependencies
+
+      - name: Download null errors file for main branch
+        uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+        with:
+          name: null_errors_main
+
+      - name: Download null errors file for PR
+        uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+        with:
+          name: null_errors_pr
+
+      - name: Download unused exports for main
```

Can we add the TODO Comment to remove this when the time comes here as well?

```suggestion
        # TODO(STENCIL-454): Remove or change this up once we've eliminated unused exports
```

(I guess-timated on the leading spacing there)